### PR TITLE
refactor(protocol): move integration/unit tests to their own folders

### PR DESCRIPTION
**Pull request by dantaik**
Reorganize uint tests and integration tests into their own folders. For certain unit tests, we would like to have the flexibility to change certain protocol constant values in LibConstants.sol, we also organized different unit tests into sub-folders. For selected tests (e.g., TaikoL1.test.ts), we can assign a dotenv file to override the constants in LibConstants.sol to deploy "customized" version of the protocol.

Note that a "customized" protocol can be deployed per npm command, not per test case or test file. As of now, all files in `test/uint/protocol` share the same custom deployment, as show in the package.json file.

```
"test:protocol": "DOTENV_CONFIG_PATH=.env.protocol pnpm hardhat test test/unit/protocol/**.test.ts",
```

File `.env.protocol` contains the overriding constants:

```
K_SKIP_PROOF_VALIDATION=true
```

### CODE REVIEW

File: .github/workflows/solidity.yml [hunk 1]
```diff
@@ -46,7 +46,11 @@ jobs:
 
       - name: protocol - Unit Tests
         working-directory: ./packages/protocol
-        run: pnpm install && pnpm clean && pnpm test:coverage && pnpm test:integration && pnpm export:abi
+        run: pnpm install && pnpm clean && pnpm test:unit
```
Comment (d1onys1us) [inline .github/workflows/solidity.yml:49]: I believe we will also need `protocol - Protocol Tests` as well here.

File: packages/protocol/.env.protocol [hunk 2]
```diff
@@ -0,0 +1 @@
+K_SKIP_PROOF_VALIDATION=true
```
Comment (d1onys1us) [inline packages/protocol/.env.protocol:1]:
```suggestion
K_SKIP_PROOF_VALIDATION=true

```

Comment (d1onys1us) [reply packages/protocol/.env.protocol:1]: also we have .env.example as well. can we combine them?

File: packages/protocol/package.json [hunk 3]
```diff
@@ -10,12 +10,11 @@
     "lint:sol": "pnpm prettier '**/*.sol' --write && pnpm solhint 'contracts/**/*.sol' --fix",
     "eslint": "pnpm exec eslint --ignore-path .eslintignore --ext .js,.ts .",
     "eslint:fix": "pnpm exec eslint --ignore-path .eslintignore --ext .js,.ts . --fix",
-    "test": "pnpm hardhat test --grep '^[^integration]'",
-    "coverage": "pnpm hardhat coverage --solcoverjs ./.solcover.js --show-stack-traces",
-    "test:coverage": "pnpm test && pnpm coverage",
+    "test:unit": "pnpm hardhat coverage --solcoverjs ./.solcover.js --show-stack-traces --testfiles 'test/unit/bridge/*.ts test/unit/bridge/**/*.test.ts test/unit/common/*.test.ts test/unit/L1/*.test.ts test/unit/L2/*.test.ts test/unit/libs/*.test.ts test/unit/thirdparty/*.test.ts'",
+    "test:protocol": "DOTENV_CONFIG_PATH=.env.protocol pnpm hardhat coverage --solcoverjs ./.solcover.js --show-stack-traces --testfiles 'test/unit/protocol/*.test.ts'",
```
Comment (d1onys1us) [inline packages/protocol/package.json:14]: why do we need these separate? can we combine them? it feels odd to have unit/L1 and unit/L2 folder, and also a unit/protocol folder with TaikoL1.test file

Comment (dantaik) [reply packages/protocol/package.json:14]: Let me reconsider the whole PR. Marking it as draft for now.

**Pull request closed without merging**